core/src/main/java/io/questdb/cairo/meta/MetaFileReader.java

```diff
@@ -0,0 +1,336 @@
+        private long currentBlockOffset;
+
+        public boolean hasNext() {
+            if (blockCount > 0 && currentBlockOffset < blocksLimit) {
+                final int blockLength = memory.getInt(currentBlockOffset + BLOCK_LENGTH_OFFSET);
+                return blockLength > 0 && currentBlockOffset + blockLength <= blocksLimit;
+            }
+            return false;
+        }
+
+        public ReadableBlock next() {
+            final int blockLength = memory.getInt(currentBlockOffset + BLOCK_LENGTH_OFFSET);
+            final short type = memory.getShort(currentBlockOffset + BLOCK_TYPE_OFFSET);
+            final byte version = memory.getByte(currentBlockOffset + BLOCK_VERSION_OFFSET);
+            final byte flags = memory.getByte(currentBlockOffset + BLOCK_FLAGS_OFFSET);
+
+            block.of(blockLength, version, flags, type, currentBlockOffset);
+            currentBlockOffset += blockLength;
+            blockCount -= 1;
+
+            return block;
+        }
+
+        public void of(int blockCount, long blocksOffset, long blocksLength) {
+            this.blockCount = blockCount;
+            this.blocksLimit = blocksOffset + blocksLength;
+            this.currentBlockOffset = blocksOffset;
+        }
+
+        class Block implements ReadableBlock {
+            private byte flags;
+            private int length;
+            private long payloadOffset;
+            private short type;
+            private byte version;
+
+            @Override
+            public long addressOf(long offset) {
+                assert offset >= 0 && offset < length;
+                return memory.addressOf(payloadOffset + offset);
+            }
+
+            @Override
+            public byte flags() {
+                return flags;
+            }
+
+            @Override
+            public BinarySequence getBin(long offset) {
+                assert offset >= 0 && offset < length;
+                return memory.getBin(payloadOffset + offset);
+            }
+
+            @Override
+            public long getBinLen(long offset) {
+                assert offset >= 0 && offset < length;
+                return memory.getBinLen(payloadOffset + offset);
+            }
+
+            @Override
+            public boolean getBool(long offset) {
+                assert offset >= 0 && offset < length;
+                return memory.getBool(payloadOffset + offset);
+            }
+
+            @Override
+            public byte getByte(long offset) {
+                assert offset >= 0 && offset < length;
+                return memory.getByte(payloadOffset + offset);
+            }
+
+            @Override
+            public char getChar(long offset) {
+                assert offset >= 0 && offset < length;
+                return memory.getChar(payloadOffset + offset);
+            }
+
+            @Override
+            public double getDouble(long offset) {
+                assert offset >= 0 && offset < length;
```

Comment:
nit: the assertions could be more precise
```suggestion
                assert offset >= 0 && offset < length - 8;
```